### PR TITLE
Catch sync failure on manual refresh

### PR DESCRIPTION
**Pull request by jonyardley**
This PR attempts to fix the issue by where a user might attempt to refresh the events list whilst under bad or non-existant network connectivity.

The sync method did not cater for any failures, therefore, the state of the events list remained in a loading state indefinitely. 

This fix handles any errors with the `.sync` method for the event refresh.

Fixes: #326
Risk Areas: Events listing

## Test Plan
Follow the bug replication steps in the github issue (linked to above)

## Pre-flight check-list

Before raising a pull request

* [ ] ~~Documentation~~
* [x] Unit tests

## Pre-merge check-list

* [x] Link to Trello ticket/GitHub issue (If applicable)
* [x] Any risky areas identified
* [x] Test plan provided
* [ ] Tester approved

## Tester check-list

* [ ] Tested on multiple devices / OS versions (Test on the physical device(s) you have access to, otherwise use BrowserStack):

  * [ ] Galaxy S8 (Android 7.0/8.0)
  * [ ] Galaxy S7 (Android 6.0)
  * [ ] iPhone X (iOS 11.x)
  * [ ] iPhone 8 (11.x)
  * [ ] iPhone 7 (iOS 10.x)

  Optional:

  * [ ] Google Pixel 2
  * [ ] Galaxy S8 Plus
  * [ ] Galaxy S7 Edge
  * [ ] Galaxy S6
  * [ ] iPhone 7+
  * [ ] iPhone 6

Accessibility Testing (If applicable)

* [ ] Text-to-Voice (Android: Voice Assistant / iOS: Speak Selection)
* [ ] Large Text (=<200%)
* [ ] Landscape Screen orientation

Weak connection testing (If applicable)

* [ ] Airplane mode
* [ ] 2G/3G Network Settings

### CODE REVIEW

File: src/integrations/cms.js [hunk 1]
```diff
@@ -68,7 +68,12 @@ export const updateCmsData = async (
         resolveLinks: false
       };
 
-  const cmsData = await clientObj.sync(syncOpts);
+  let cmsData;
+  try {
+    cmsData = await clientObj.sync(syncOpts);
+  } catch (e) {
+    return { ...localCmsData, updated: false };
+  }
```
Comment (jonyardley) [inline src/integrations/cms.js]: I have assumed the try/catch pattern is ok here?

Comment (jonyardley) [reply src/integrations/cms.js]: Not sure if we want to apply the same pattern here as well? https://github.com/redbadger/pride-london-app/pull/357/files#diff-55210a6e2d724432b6b63cbc601e4479R42

**Comment by jonyardley** [comment]
@frigus02 Totally agree. Just had a chat with @clmntnbrn about a notification design for this and other runtime errors.

**Comment by RGBboy** [comment]
If we decide to go down the notify path, it may be a good idea to represent our state using a RemoteData pattern: http://blog.jenkster.com/2016/06/how-elm-slays-a-ui-antipattern.html

File: src/integrations/cms.js (hunk 1 above)
Comment (charypar) [reply src/integrations/cms.js]: Yea, actually, we probably should, since we can recover from that failure

**Comment by frigus02** [comment]
I'm going to move this try-catch to the reducer.

**Comment by charypar** [comment]
@frigus02 Should we close this PR then?

**Comment by charypar** [comment]
Decided to close with @frigus02 as the changes would get undone anyway.

**Pull request closed without merging**